app/models/company.rb

```diff
@@ -2,4 +2,28 @@ class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
   has_many :jobs
+  has_many :contacts
+
+  def self.by_city(desired_city)
+    Company.where(city: desired_city)
+  end
+
+  def self.top_3_by_average_interest
```

Comment:
this method is doing two separate pieces of logic. can we break this method into two or even three?